src/main/java/com/khoros/twitter/services/TwitterLabService.java

```diff
@@ -58,23 +58,33 @@ public Status postTweet(String message) throws TwitterLabException {
 
     public List<Post> getTimeline() throws TwitterLabException {
         LOGGER.info("Attempting to retrieve home timeline from cache.");
-        if (cache.size() == 0) {
-            updateCache();
+        if (!cache.containsKey("timeline") || cache.get("timeline").size() == 0) {
+            updateCache("timeline");
         }
-        List<Post> postTimeline = new ArrayList<Post>(cache.values());
+        List<Post> postTimeline = cache.get("timeline");
         LOGGER.info("Successfully retrieved home timeline from cache.");
         return postTimeline;
     }
 
+    public List<Post> getUserTimeline() throws TwitterLabException {
+        LOGGER.info("Attempting to retrieve user timeline from cache.");
+        if (!cache.containsKey("userTimeline") || cache.get("userTimeline").size() == 0) {
```

Comment:
i'd recommend having this method check the cache and actually pull the information from twitter.  Use the updateCache method to convert/cache as more of a helper method on the data.